tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs

```diff
@@ -447,82 +532,189 @@ public static void TestSpanDangerousGetPinnableReferenceByte(int length)
-            var span = new Span<string>(array);
-            foreach (var iteration in Benchmark.Iterations)
-                using (iteration.StartMeasurement())
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                    {
-                        ref string temp = ref span.DangerousGetPinnableReference();
-                    }
+            InvokeTestSpanDangerousGetPinnableReference<string>(length);
+        }
+
+        static void InvokeTestSpanDangerousGetPinnableReference<T>(int length)
+        {
+            var array = new T[length];
+
+            Invoke((int innerIterationCount) => TestSpanDangerousGetPinnableReference<T>(array, innerIterationCount),
+                "TestSpanDangerousGetPinnableReference<{0}>({1})", typeof(T).Name, length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void TestSpanDangerousGetPinnableReference<T>(T[] array, int iterationCount)
+        {
+            var sink = Sink<T>.Instance;
+            var span = new Span<T>(array);
+
+            for (int i = 0; i < iterationCount; i++)
+            {
+                ref T temp = ref span.DangerousGetPinnableReference();
+                sink.Data = temp;
+            }
         }
         #endregion
 
-        #region TestSpanIndex<T>
+        #region TestSpanIndexHoistable<T>
         [Benchmark(InnerIterationCount = BaseIterations)]
         [InlineData(1)]
         [InlineData(10)]
         [InlineData(100)]
         [InlineData(1000)]
-        public static void TestSpanIndexByte(int length)
+        public static void TestSpanIndexHoistableByte(int length)
         {
-            var array = new byte[length];
-            var span = new Span<byte>(array);
-            byte temp;
-            foreach (var iteration in Benchmark.Iterations)
-                using (iteration.StartMeasurement())
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                            temp = span[length/2];
+            InvokeTestSpanIndexHoistable<byte>(length);
+        }
+
+        [Benchmark(InnerIterationCount = BaseIterations)]
+        [InlineData(1)]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(1000)]
+        public static void TestSpanIndexHoistableString(int length)
+        {
+            InvokeTestSpanIndexHoistable<string>(length);
+        }
+
+        static void InvokeTestSpanIndexHoistable<T>(int length)
+        {
+            var array = new T[length];
+
+            Invoke((int innerIterationCount) => TestSpanIndexHoistable<T>(array, length, innerIterationCount),
+                "TestSpanIndexHoistable<{0}>({1})", typeof(T).Name, length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void TestSpanIndexHoistable<T>(T[] array, int length, int iterationCount)
+        {
+            var sink = Sink<T>.Instance;
+            var span = new Span<T>(array);
```

Comment:
Isn't this test also measuring the construction of Span from an array and not just indexer? When compared to the array indexer this would give worst results.